classes/ObjectModel.php

```diff
@@ -2147,9 +2147,9 @@ public function getFieldByLang($field_name, $id_lang = null)
      *
      * @since 1.5.0.1
      *
-     * @param array $fields
+     * @param array<string, bool|array<int, bool>>|null $fields
      */
-    public function setFieldsToUpdate(array $fields)
+    public function setFieldsToUpdate(?array $fields)
```

Comment:
It's a BC. What is someone use `setFieldsToUpdate` without calling `addFieldsToUpdate`? 
You must keep the original behaviour and don't call `setFieldsToUpdate` when there is nothing to add
```suggestion
    public function setFieldsToUpdate(array $fields)
```

Reply:
as discussed in slack- its OK, because initial update_fields value is NULL.

Reply:
Yep it's not a BC since the default value is `null` And it doesn't prevent from using the method with an array